propagation/propagation.go

```diff
@@ -55,14 +68,19 @@ func (c MapCarrier) Keys() []string {
 	return keys
 }
 
-// HeaderCarrier adapts http.Header to satisfy the TextMapCarrier interface.
+// HeaderCarrier adapts http.Header to satisfy the TextMapCarrier and ValuesGetter interfaces.
```

Comment:
You can also add

```go
// Compile time check that HeaderCarrier implements the ValuesGetter.
var _ ValuesGetter = HeaderCarrier{}
```

Reply:
For consistency, I will add for both types: 

```go
// Compile time check that HeaderCarrier implements the ValuesGetter.
var _ TextMapCarrier = HeaderCarrier{}

// Compile time check that HeaderCarrier implements the TextMapCarrier.
var _ ValuesGetter = HeaderCarrier{}
```